lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -134,6 +135,56 @@ MObject MayaUsdProxyShapeBase::outTimeAttr;
 MObject MayaUsdProxyShapeBase::outStageDataAttr;
 MObject MayaUsdProxyShapeBase::outStageCacheIdAttr;
 
+namespace {
+// utility function to extract the tag name from an anonymous layer.
+// e.g
+// given layer identifier = anon:00000232FE3FB470:anonymousLayer1234
+// tag name = anonymousLayer1234
+std::string extractAnonTagName(const std::string& identifier)
+{
+    std::size_t found = identifier.find_last_of(":");
+    return identifier.substr(found + 1);
+}
+
+// recursive function to create new anonymous Sublayer(s)
+// and set the edit target accordingly.
+void createNewAnonSubLayerRecursive(
+    UsdStageRefPtr newUsdStage,
+    SdfLayerRefPtr targetLayer,
+    SdfLayerRefPtr parentLayer)
```

Comment:
Minor: shouldn't we pass these pointers in as const-ref, to avoid creating the temporary argument pointers and therefore avoiding an inc / dec of the ref count?

Reply:
Thanks @ppt-adsk. All addressed: ad23ef814853ca48d0f7b3be0970cde9231fe6bc